python/cuml/benchmark/nvtx_benchmark.py

```diff
@@ -16,16 +16,22 @@
 
 import os
 import sys
+import tempfile
 from subprocess import run
 import json
 
 
 class Profiler:
-    def __init__(self, tmp_path="/tmp/nsys_report"):
-        self.nsys_file = tmp_path + "/report.nsys-rep"
-        self.json_file = tmp_path + "/report.json"
-        self._execute(["rm", "-rf", tmp_path])
-        self._execute(["mkdir", "-p", tmp_path])
+    def __init__(self, tmp_path=None):
+        self.tmp_dir = tempfile.TemporaryDirectory(dir=tmp_path)
+        self.nsys_file = os.path.join(self.tmp_dir.name, "report.nsys-rep")
+        self.json_file = os.path.join(self.tmp_dir.name, "report.json")
+        self._execute(["rm", "-rf", self.tmp_dir.name])
+        self._execute(["mkdir", "-p", self.tmp_dir.name])
```

Comment:
One additional improvement would be to call the Python builtin functions for these. Submitted PR ( https://github.com/rapidsai/cuml/pull/5537 ) doing just that